internal/receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -0,0 +1,150 @@
+
+	"github.com/signalfx/splunk-otel-collector/internal/receiver/prometheusremotewritereceiver/internal"
+)
+
+var _ receiver.Metrics = (*simplePrometheusWriteReceiver)(nil)
+
+// simplePrometheusWriteReceiver implements the receiver.Metrics for PrometheusRemoteWrite protocol.
+type simplePrometheusWriteReceiver struct {
+	server       *internal.PrometheusRemoteWriteServer
+	reporter     internal.Reporter
+	nextConsumer consumer.Metrics
+	cancel       context.CancelFunc
+	config       *Config
+	settings     receiver.CreateSettings
+	sync.Mutex
+}
+
+// newPrometheusRemoteWriteReceiver creates the PrometheusRemoteWrite receiver with the given parameters.
+func newPrometheusRemoteWriteReceiver(
+	settings receiver.CreateSettings,
+	config *Config,
+	nextConsumer consumer.Metrics,
+) (*simplePrometheusWriteReceiver, error) {
+	if nextConsumer == nil {
+		return nil, component.ErrNilNextConsumer
+	}
+
+	rep, err := newReporter(settings)
+	if err != nil {
+		return nil, err
+	}
+
+	r := &simplePrometheusWriteReceiver{
+		settings:     settings,
+		config:       config,
+		nextConsumer: nextConsumer,
+		reporter:     rep,
+	}
+	return r, nil
+}
+
+// Start starts an HTTP server that can process Prometheus Remote Write Requests
+func (receiver *simplePrometheusWriteReceiver) Start(ctx context.Context, host component.Host) error {
+	metricsChannel := make(chan pmetric.Metrics, receiver.config.BufferSize)
+	cfg := &internal.ServerConfig{
+		HTTPServerSettings: receiver.config.HTTPServerSettings,
+		Path:               receiver.config.ListenPath,
+		Mc:                 metricsChannel,
+		Reporter:           receiver.reporter,
+		Host:               host,
+	}
+	ctx, receiver.cancel = context.WithCancel(ctx)
+	server, err := internal.NewPrometheusRemoteWriteServer(ctx, cfg)
+	if err != nil {
+		return err
+	}
+	receiver.Lock()
+	defer receiver.Unlock()
+	if nil != receiver.server {
+		err := receiver.server.Close()
+		if err != nil {
+			return err
+		}
+	}
+	receiver.server = server
+
+	go receiver.startServer(host)
+	go receiver.manageServerLifecycle(ctx, metricsChannel)
+
+	return nil
+}
+
+func (receiver *simplePrometheusWriteReceiver) startServer(host component.Host) {
+	receiver.Lock()
+	server := receiver.server
+	if server == nil {
+		host.ReportFatalError(fmt.Errorf("start called on null server for receiver %s", typeString))
+	}
+	receiver.Unlock()
+	if err := server.ListenAndServe(); err != nil {
+		host.ReportFatalError(err)
+		receiver.reporter.OnDebugf("Error in %s/%s listening server %s: %s", typeString, receiver.settings.ID, server.Addr, err)
+	}
+}
+
+func (receiver *simplePrometheusWriteReceiver) manageServerLifecycle(ctx context.Context, metricsChannel chan pmetric.Metrics) {
+	for {
+		select {
+		case metrics := <-metricsChannel:
+			receiver.Lock()
+			metricContext := receiver.reporter.StartMetricsOp(ctx)
+			err := receiver.flush(metricContext, metrics)
+			if err != nil {
+				receiver.reporter.OnTranslationError(metricContext, err)
```

Comment:
the err variable here is not related to a translation error, it seems related to a flushing error.